tests/plotting/test_plot_manager.py

```diff
@@ -406,3 +407,57 @@ async def test_plot_info_caching(test_environment):
     await run_refresh_test(plot_manager)
     assert len(plot_manager.plots) == len(plot_manager.plots)
     plot_manager.stop_refreshing()
+
+
+@pytest.mark.parametrize(
+    ["event_to_raise"],
+    [
+        pytest.param(PlotRefreshEvents.started, id="started"),
+        pytest.param(PlotRefreshEvents.batch_processed, id="batch_processed"),
+        pytest.param(PlotRefreshEvents.done, id="done"),
+    ],
+)
+@pytest.mark.asyncio
+async def test_callback_event_raises(test_environment, event_to_raise: PlotRefreshEvents):
+    last_even_fired: Optional[PlotRefreshEvents] = None
+
+    def raising_callback(event: PlotRefreshEvents, _: PlotRefreshResult):
+        nonlocal last_even_fired
+        last_even_fired = event
+        if event == event_to_raise:
+            raise Exception(f"run_raise_in_callback {event_to_raise}")
+
+    env: TestEnvironment = test_environment
+    # Load dir_1
+    add_plot_directory(env.root_path, str(env.dir_1.path))
+    expected_result.loaded = len(env.dir_1)
+    expected_result.removed = 0
+    expected_result.processed = len(env.dir_1)
+    expected_result.remaining = 0
+    await run_refresh_test(env.plot_manager)
+    # Load dir_2
+    add_plot_directory(env.root_path, str(env.dir_2.path))
+    expected_result.loaded = len(env.dir_2)
+    expected_result.removed = 0
+    expected_result.processed = len(env.dir_1) + len(env.dir_2)
+    expected_result.remaining = 0
+    await run_refresh_test(env.plot_manager)
+    # Now raise the exception in the callback
+    default_callback = env.plot_manager._refresh_callback
+    env.plot_manager.set_refresh_callback(raising_callback)
+    await run_refresh_test(env.plot_manager, expect_match=False)
+    # And make sure the follow-up evens aren't fired
+    assert last_even_fired is not None and last_even_fired.value == event_to_raise.value
```

Comment:
> Can we not just check equality?

Wouldn't this mean we need to initialize `last_event_fired` with the actual expected result beforehand in one case? Its meant to be a "make sure it was even triggered" check.

Reply:
I don't believe this suggested change has any effect on how the variable would be initialized.  I'm not clear what case you are concerned about that would cause the simple equality check to do the wrong thing.  `None == PlotsRefreshEvents.started` is a valid thing to do and will evaluate to `False`.

Reply:
Right, i obviously confused myself. See 7053f608d78281ad346cc5827348776174ba83b1.